apps/designer/app/designer/features/topbar/sync-status.tsx

```diff
@@ -1,50 +1,62 @@
 import { useStore } from "@nanostores/react";
 import {
-  theme,
   Flex,
-  keyframes,
   AccessibleIcon,
-  styled,
-  Box,
+  rawTheme,
+  Tooltip,
 } from "@webstudio-is/design-system";
-import { CheckIcon, DotsHorizontalIcon } from "@webstudio-is/icons";
-import { syncStatus } from "~/designer/shared/sync-server";
+import { CloudIcon } from "@webstudio-is/icons";
+import { useEffect, useState } from "react";
+import { syncStatus } from "~/designer/shared/sync";
 
-const iconSize = 15;
-
-const StyledCheckIcon = styled(CheckIcon, {
-  width: iconSize,
-  height: iconSize,
-  background: theme.colors.green9,
-  borderRadius: theme.borderRadius.round,
-});
-
-const ellipsisKeyframes = keyframes({
-  to: { width: iconSize },
-});
-
-const AnimatedDotsIcon = () => {
-  return (
-    <Flex direction="column" css={{ width: iconSize }}>
-      <Box
-        css={{
-          animation: `${ellipsisKeyframes} steps(4,end) 900ms infinite`,
-          width: 0,
-          overflow: "hidden",
-        }}
-      >
-        <DotsHorizontalIcon width="12" height="12" />
-      </Box>
-    </Flex>
-  );
+const useIsOnline = () => {
+  const [isOnline, setIsOnline] = useState(false);
+  useEffect(() => {
+    const handle = () => setIsOnline(navigator.onLine);
+    addEventListener("offline", handle);
+    addEventListener("online", handle);
+    return () => {
+      removeEventListener("offline", handle);
+      removeEventListener("online", handle);
+    };
+  }, []);
+  return isOnline;
```

Comment:
```
const subscribeOnline = (cb) => {
  addEventListener("offline", handle);
  addEventListener("online", handle);
  return () => {
    removeEventListener("offline", handle);
    removeEventListener("online", handle);
  };
}

const isOnline = () => navigator.onLine

const isOnlineServer = () => true

useSyncExternalStore(subscribeOnline, isOnline, isOnlineServer)
```

Reply:
it shouldn't be used server-side at all, neat notation but a bit of indirection

Reply:
It's optional. You can not use it if component is client only